### ISSUE TITLE
Alternative Losses

### ISSUE THREAD

**Issue by ClashLuke** [HomebrewNLP/HomebrewNLP-Jax#54]
Currently, we're using only the softmax classification/cross-entropy loss to create a language-modeling loss for next-token prediction. However, other works such as [T-Few](https://arxiv.org/abs/2205.05638) showed that adding alternative losses for external benefits such as length explicit penalties during training can help downstream-task performance. Additionally, other works like [DCL](https://arxiv.org/abs/2110.06848) and [InfoLOOB](https://arxiv.org/abs/1905.06922) demonstrated that changing the fundamental structure of the loss from softmax classification to something different can help speed up convergence. That's why a similar approach could be beneficial for us.\
In this issue, we'll explore whether InfoLOOB's classification loss for the language-modeling objective helps or if we should change the entire objective.

**Comment by ClashLuke** [comment]
[PolyLoss](https://wandb.ai/homebrewnlp/gpt/runs/3iplr5zq) vs [CrossEntropy](https://wandb.ai/homebrewnlp/gpt/runs/6qf300ff)

**Comment by ClashLuke** [comment]
PolyLoss (green) performs quite a bit worse than CrossEntropy (bisque):
![grafik](https://user-images.githubusercontent.com/39779310/171992425-70c428d7-63b6-46d3-8461-846c26999419.png)

We could still try InfoLOOB (DCL) as it appeared promising before:
![grafik](https://user-images.githubusercontent.com/39779310/171992481-397e02fc-9441-4fe3-b6bb-43f986e81314.png)
However, after reaching a loss of -100, InfoLOOB ran into NaN, which halted the training. Nothing like this has happened with CrossEntropy, which is why CrossEntropy achieved a better final model even though the initial convergence was slower.
